gcsmigrate/shared.py

```diff
@@ -0,0 +1,354 @@
+"""Tools for migrating Mogile content to GCS."""
+
+import base64
+import hashlib
+import io
+import json
+import os
+import random
+import shutil
+import time
+
+import dj_database_url
+import pymysql
+import requests
+
+BUFSIZE = 16 * 1024
+
+
+class HashWrap(io.RawIOBase):
```

Comment:
This is cool!

Reply:
Thanks! I was trying to not write the content to disk, because GCF uses ram tmp disk (so it counts against our memory usage). I'm not sure if some of the libraries do use disk, though, because I had to increase the memory size. In any case, it seems to work and was fun to write.

Reply:
Ah, that makes sense. Very clever solution!